client/internal/user/signin.go

```diff
@@ -0,0 +1,38 @@
+package user
+
+import (
+	"client/internal/current"
+	"client/internal/logger"
+	"client/internal/models"
+	"context"
+	"database/sql"
+)
+
+func (i *Item) SignIN(f models.User) error {
+
+	u, err := i.storage.UserGet(context.Background(), f.Login)
+	if err == nil {
+		logger.Logger.Info("user athentificated locally")
+		u.Password = f.Password
+
+		current.SetUser(u)
+		return nil
+	}
+	if err == sql.ErrNoRows {
+		logger.Logger.Info("try to athentificate user by server")
+		u, roerr := i.roamer.UserGet(f)
+		if roerr == nil {
+			err = i.storage.UserCreate(context.Background(), u)
+			if err != nil {
+				logger.Logger.Error("when create user locally", "err", err)
+				return err
+			}
+			u.Password = f.Password
+
+			current.SetUser(u)
+			return nil
+		}
+		return roerr
```

Comment:
Здесь то же самое - проверяем, произошла ли ошибка и быстро возвращаем. Это легче для воспрития ветвления логики, особенно, если этот подход используется везде. Так же - ошибка (или ее отсутсвие) не длолжна управлять нашей бизнес логикой. Ошибка значит, что что-то пошло не так и наша задача проверить это как можно быстрее и вренуть управление.
```suggestion
                logger.Logger.Info("try to athentificate user by server")
		u, roerr := i.roamer.UserGet(f)
		if roerr != nil {
			return roerr
		}

		err = i.storage.UserCreate(context.Background(), u)
		if err != nil {
			logger.Logger.Error("when create user locally", "err", err)
			return err
		}
		u.Password = f.Password

		current.SetUser(u)
		return nil
		return roerr
```

Reply:
Да, такой подход более явный
Исправил